### DISCUSSION TITLE
Plink out of memory error during ancestry match combine step

### DISCUSSION THREAD

**Discussion by h171e** [PGScatalog/pgsc_calc#205] [Q&A] [answered]
Got this error after increasing available ram to 7 Gb installed in the computer and updating the config to use up to 7 Gb of ram. 

I wonder if there is an indication for required ram in the error message or if its due to required physical storage memory. 

Last lines of the error message: 

**Error: Out of memory.  The --memory flag may be helpful.
  Failed allocation size: 1011902592**

Last portion of the error message:

 **Start time: Thu Oct 26 22:08:30 2023
  7888 MiB RAM detected; reserving 7168 MiB for main workspace.
  Using up to 2 compute threads.
  3942 samples (1860 females, 2082 males; 3942 founders) loaded from
  GRCh38_HGDP+1kGP_ALL.psam.
  66505870 out of 84325213 variants loaded from GRCh38_HGDP+1kGP_ALL.pvar.zst.
  3 categorical phenotypes loaded.
  --extract: 49558808 variants remaining.
  Note: Skipping --rm-dup since no duplicate IDs are present.
  --remove: 3330 samples remaining.
  Calculating sample missingness rates... done.
  0 samples removed due to missing genotype data (--mind).
  3330 samples (1564 females, 1766 males; 3330 founders) remaining after main
  filters.

  Error: Out of memory.  The --memory flag may be helpful.
  Failed allocation size: 1011902592
  End time: Thu Oct 26 22:10:05 2023**

![Screenshot_2023_10_26-2](https://github.com/PGScatalog/pgsc_calc/assets/139821593/17237fe4-87a0-48e3-8e93-b716efc46dd9)

**Comment by nebfield** [comment, marked as answer]
[You'll need to allocate more memory to that process](https://pgsc-calc.readthedocs.io/en/latest/how-to/bigjob.html#configuring-pgsc-calc-to-use-more-resources-locally), maybe try 16 GB.
